jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/DataUtils.scala

```diff
@@ -180,4 +180,43 @@ object DataUtils extends Serializable {
     repartitionRDDs(deterministicPartition, numWorkers, arrayOfRDDs)
   }
 
+  private[spark] def convertDataFrameToXGBLabeledPointRDDs(
+      labelCol: Column,
+      featuresCols: Seq[Column],
+      weight: Column,
+      baseMargin: Column,
+      group: Option[Column],
+      numWorkers: Int,
+      deterministicPartition: Boolean,
+      dataFrames: DataFrame*): Array[RDD[XGBLabeledPoint]] = {
+    // Order is [label, weight, margin, features, group]
```

Comment:
how we prevent users messing up by passing an unexpected schema?

Reply:
Actually the order in this comment is defined by us via `DataFrame.select` below @line 206. Not meaning the order provided in schema by users.

We also has some check for the schema [here](https://github.com/dmlc/xgboost/blob/47444e1fb9fa53950e79276726f5d4069e750c9a/jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostEstimatorCommon.scala#L35)